core/server/automations_test.go

```diff
@@ -49,6 +51,66 @@ func TestListKustomizations(t *testing.T) {
 	g.Expect(res.Kustomizations[0].Name).To(Equal(appName))
 }
 
+func TestGetKustomization(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	ctx := context.Background()
+
+	c, cleanup := makeGRPCServer(k8sEnv.Rest, t)
+	defer cleanup()
+
+	_, k, err := kube.NewKubeHTTPClientWithConfig(k8sEnv.Rest, "")
+	g.Expect(err).NotTo(HaveOccurred())
+
+	appName := "myapp"
+	ns := newNamespace(ctx, k, g)
+
+	kust := &kustomizev1.Kustomization{
+		Spec: kustomizev1.KustomizationSpec{
+			SourceRef: kustomizev1.CrossNamespaceSourceReference{
+				Kind: "GitRepository",
+			},
+		},
+		Status: kustomizev1.KustomizationStatus{
+			Inventory: &kustomizev1.ResourceInventory{
+				Entries: []kustomizev1.ResourceRef{
+					{
+						Version: "v1",
+						ID:      ns.Name + "_my-deployment_apps_Deployment",
+					},
+				},
+			},
+		},
+	}
+	kust.Name = appName
+	kust.Namespace = ns.Name
+
+	g.Expect(k.Create(ctx, kust)).To(Succeed())
+
+	t.Run("gets a kustomization", func(t *testing.T) {
+		_, err = c.GetKustomization(ctx, &pb.GetKustomizationRequest{Name: appName, Namespace: ns.Name})
+		g.Expect(err).NotTo(HaveOccurred())
+
+		// The kustomization status does not get populated when running in a test environment,
+		// probably because there is no kustomization-controller running to populate it.
+		// Even though we populate it in the kustomization.Status object above, it still does not populate.
+		// Need some help from the flux team on this.
```

Comment:
do we care about those status bits? as in, is that something we would add/control anyway or is that something the flux controller would add? if it is a flux thing, then I think we are find not checking those things here (we can do that in an e2e) and just check that the result name is the right one because that is behaviour that we control.

Reply:
In order for the endpoint to be "working" from the perspective of the UI, we need the status of the Kustomization to be populated.

If we can simulate *without* running a Flux reconciler in the test environment, that would be ideal. @hiddeco Any tricks for getting this status populated?

Reply:
The problem I think, is with the usage of `Create` which does not persist the Status. First creating the object, and then patching it with a Status should work.